### PR TITLE
fix finish_reason

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

File: lmdeploy/serve/async_engine.py [hunk 1]
```diff
@@ -243,9 +243,12 @@ async def generate(
 
                 # `response_size` might be note updated since
                 # ` if response.endswith('�')`
-                if response_size != tokens:
-                    yield GenOut(response, self.steps[str(session_id)],
-                                 len(input_ids), tokens, finish_reason)
+                finish_reason = 'length' \
+                    if tokens >= request_output_len else 'stop'
+                if response_size == tokens:
```
Comment (lvhan028) [inline lmdeploy/serve/async_engine.py:248]: What's the motivation to set `response=""` when `response_size ==  tokens`

Comment (AllentDan) [reply lmdeploy/serve/async_engine.py:248]: we have to return response if `response_size !=  tokens`. Nothing special, just make sure we do not return the last response twice.

**Review by lvhan028 — APPROVED** [review]
LGTM

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**